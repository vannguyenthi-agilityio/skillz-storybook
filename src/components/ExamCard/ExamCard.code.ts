export const asDefault = `
import ExamCard from '@components/ExamCard';

<ExamCard
  author="Jame Rodriguez"
  color="cyan"
  dateTime="8:00 AM - 21 Dec, 2022"
  enroller={1024}
  price={2}
  score={50}
  title="Card title will goes here"
/>
`;

export const failedCard = `
import ExamCard from '@components/ExamCard';

<ExamCard
  color="purple"
  dateTime="8:00 AM - 21 Dec, 2022"
  maxScore={90}
  order={4}
  score={20}
  status="failed"
  title="Card title will goes here"
/>
`;

export const newFailedCard = `
import ExamCard from '@components/ExamCard';

<ExamCard
  color="purple"
  dateTime="8:00 AM - 21 Dec, 2022"
  isNew
  maxScore={90}
  order={9}
  score={20}
  status="failed"
  title="Card title will goes here"
/>
`;

export const successCard = `
import ExamCard from '@components/ExamCard';

<ExamCard
  color="orange"
  dateTime="8:00 AM - 21 Dec, 2022"
  maxScore={90}
  order={11}
  score={70}
  status="success"
  title="Card title will goes here"
/>
`;

export const newSuccessCard = `
import ExamCard from '@components/ExamCard';

<ExamCard
  color="orange"
  dateTime="8:00 AM - 21 Dec, 2022"
  isNew
  maxScore={100}
  order={18}
  score={85}
  status="success"
  title="Card title will goes here"
/>
`;
