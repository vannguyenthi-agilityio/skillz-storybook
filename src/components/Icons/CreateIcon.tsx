export const CreateIcon = () => {
  return (
    <svg
      width='17'
      height='17'
      viewBox='0 0 17 17'
      fill='none'
      xmlns='http://www.w3.org/2000/svg'
    >
      <path
        d='M7.68748 0.375011H9.40998V7.68751H16.7225V9.41001H9.40998V16.7225H7.68748V9.41001H0.374985V7.68751H7.68748V0.375011Z'
        fill='black'
      />
    </svg>
  );
};
