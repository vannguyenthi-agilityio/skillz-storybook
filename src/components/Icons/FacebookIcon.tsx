// Interface
import type { IIcon } from '@interfaces/icon.interface';

export const FacebookIcon = ({ w = '34', h = '34' }: IIcon) => {
  return (
    <svg
      width={w}
      height={h}
      viewBox='0 0 34 34'
      fill='none'
      xmlns='http://www.w3.org/2000/svg'
    >
      <path
        d='M0.33667 17.0035C0.338609 25.2025 6.30067 32.1835 14.3983 33.4685V21.8202H10.17V17.0035H14.4033V13.3368C14.2141 11.5994 14.8075 9.86781 16.0227 8.61171C17.2379 7.35561 18.9489 6.70518 20.6917 6.83682C21.9426 6.85702 23.1903 6.96843 24.425 7.17016V11.2685H22.3183C21.5931 11.1735 20.8639 11.413 20.3363 11.9196C19.8086 12.4262 19.5396 13.145 19.605 13.8735V17.0035H24.2233L23.485 21.8218H19.605V33.4685C28.3625 32.0845 34.5035 24.0865 33.5792 15.2686C32.6549 6.45077 24.9888 -0.0999629 16.1347 0.362119C7.28061 0.824201 0.338077 8.13734 0.33667 17.0035Z'
        fill='#F3F4F6'
      />
    </svg>
  );
};
