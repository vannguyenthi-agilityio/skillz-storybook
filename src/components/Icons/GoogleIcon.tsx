// Interface
import type { IIcon } from '@interfaces/icon.interface';

export const GoogleIcon = ({ w = '37', h = '38' }: IIcon) => {
  return (
    <svg
      width={w}
      height={h}
      viewBox='0 0 37 38'
      fill='none'
      xmlns='http://www.w3.org/2000/svg'
    >
      <path
        d='M18.9085 15.5747V22.7643H28.8918C27.9626 27.333 24.0731 29.958 18.9085 29.958C12.8929 29.8752 8.05985 24.9752 8.05985 18.9591C8.05985 12.943 12.8929 8.0429 18.9085 7.96011C21.4089 7.95711 23.8324 8.82485 25.7626 10.4143L31.1793 4.99761C24.966 -0.464749 15.8918 -1.16732 8.91172 3.27353C1.93162 7.71439 -1.27689 16.2314 1.03816 24.1739C3.35321 32.1164 10.6355 37.5756 18.9085 37.5705C28.2147 37.5705 36.6772 30.8018 36.6772 18.958C36.6629 17.8177 36.5231 16.6824 36.2606 15.5726L18.9085 15.5747Z'
        fill='white'
      />
    </svg>
  );
};
