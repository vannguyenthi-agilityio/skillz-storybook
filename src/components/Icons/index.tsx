export * from './CharityIcon';
export * from './MetaMaskIcon';
export * from './RightArrowIcon';
export * from './AddIcon';
export * from './UnionIcon';
export * from './SearchIcon';
export * from './LogoIcon';
export * from './MaskIcon';
export * from './SignInIcon';
export * from './SlackIcon';
export * from './StarIcon';
export * from './PeopleIcon';
export * from './FailedIcon';
export * from './BellIcon';
export * from './SuccessIcon';
