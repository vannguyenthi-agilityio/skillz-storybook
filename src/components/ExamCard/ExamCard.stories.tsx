// Component
import ExamCard from '.';

// Types
import type { IExamCard } from '.';
import type { Meta, Story } from '@storybook/react/dist/ts3.9/client/preview/types-6-0';

// Enums
import { CARD_STATUS, COLORS } from '@enums';

// Code script
import { asDefault, failedCard, newFailedCard, newSuccessCard, successCard } from './ExamCard.code';

export default {
  title: 'Components/ExamCard',
  component: ExamCard,
  argTypes: {
    color: {
      options: ['cyan', 'yellow', 'orange', 'purple', 'lime'],
      control: {
        type: 'select'
      }
    },
    status: {
      options: ['success', 'failed', 'initial'],
      control: {
        type: 'inline-radio'
      }
    },
    isNew: {
      type: 'boolean',
      defaultValue: false
    }
  }
} as Meta;

const Template: Story<IExamCard> = (args) => <ExamCard {...args} />;

export const Default = Template.bind({});
Default.args = {
  color: COLORS.CYAN,
  title: 'Lorem ipsum dolor sit amet, consectetu',
  price: 2,
  author: 'Jame Rodriguez',
  dateTime: '8:00 AM - 21 Dec, 2022',
  enroller: 1024,
  score: 50,
  status: CARD_STATUS.INITIAL
};
Default.parameters = {
  componentSource: {
    code: asDefault,
    language: 'typescript'
  }
};

export const FailedCard = Template.bind({});
FailedCard.args = {
  color: COLORS.PURPLE,
  title: 'Card title will goes here',
  status: CARD_STATUS.FAILED,
  score: 20,
  maxScore: '90',
  order: 4,
  dateTime: '8:00 AM - 21 Dec, 2022'
};
FailedCard.parameters = {
  componentSource: {
    code: failedCard,
    language: 'typescript'
  }
};

export const NewFailedCard = Template.bind({});
NewFailedCard.args = {
  color: COLORS.PURPLE,
  title: 'Card title will goes here',
  status: CARD_STATUS.FAILED,
  score: 20,
  maxScore: 90,
  dateTime: '8:00 AM - 21 Dec, 2022',
  order: 9,
  isNew: true
};
NewFailedCard.parameters = {
  componentSource: {
    code: newFailedCard,
    language: 'typescript'
  }
};

export const SuccessCard = Template.bind({});
SuccessCard.args = {
  color: COLORS.ORANGE,
  status: CARD_STATUS.SUCCESS,
  title: 'Card title will goes here',
  maxScore: 90,
  score: 70,
  dateTime: '8:00 AM - 21 Dec, 2022',
  order: 11
};
SuccessCard.parameters = {
  componentSource: {
    code: successCard,
    language: 'typescript'
  }
};

export const NewSuccessCard = Template.bind({});
NewSuccessCard.args = {
  color: COLORS.ORANGE,
  status: CARD_STATUS.SUCCESS,
  title: 'Card title will goes here',
  maxScore: 100,
  score: 85,
  dateTime: '8:00 AM - 21 Dec, 2022',
  order: 18,
  isNew: true
};
NewSuccessCard.parameters = {
  componentSource: {
    code: newSuccessCard,
    language: 'typescript'
  }
};
