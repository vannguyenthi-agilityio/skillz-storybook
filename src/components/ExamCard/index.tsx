// Components
import { Card, CardHeader, CardBody, CardFooter, Text, Box, HStack, Flex, Icon } from '@chakra-ui/react';
import { BellIcon, PeopleIcon, FailedIcon, SlackIcon, StarIcon, SuccessIcon } from '@components/Icons';

// Types
import type { CardProps } from '@chakra-ui/react';

// Enums
import { CARD_STATUS, COLORS } from '@enums';

// Helpers
import { convertPxToRem } from '@helpers';

export interface IExamCard extends CardProps {
  color: string;
  isNew?: boolean;
  status: CARD_STATUS;
  title: string;
  price?: number | string;
  score?: number | string;
  author?: string;
  enroller: number | string;
  dateTime: string;
  maxScore?: string | number;
  order: string | number;
  icon: JSX.Element;
}

const ExamCard = ({
  maxW = '240px',
  title,
  price,
  color = COLORS.CYAN,
  isNew,
  status = CARD_STATUS.INITIAL,
  score,
  maxScore = 100,
  author,
  enroller,
  dateTime,
  order = 1,
  icon = <SlackIcon />,
  ...rest
}: IExamCard) => {
  const renderInitialCard = () => (
    <>
      <HStack
        mb={convertPxToRem(14)}
        mt={convertPxToRem(23)}
      >
        <Icon as={StarIcon} />
        <Text
          fontSize='20px'
          lineHeight='16px'
        >
          {price}
        </Text>
      </HStack>
      <HStack>
        <Icon
          as={PeopleIcon}
          viewBox='16px'
        />
        <Text
          as='b'
          lineHeight='16px'
        >
          {enroller}
        </Text>
        <Text
          lineHeight='16px'
          fontSize='14px'
        >
          already enrolled
        </Text>
      </HStack>
    </>
  );

  const renderFailedCard = () => (
    <>
      {isNew ? (
        <HStack
          pt='10px'
          color={`${COLORS.PINK}.650`}
        >
          <Icon as={BellIcon} />
          <Text>New</Text>
        </HStack>
      ) : null}
      <Box
        pos='absolute'
        bottom='84px'
        right='30px'
        data-testid='failed'
      >
        <Icon as={FailedIcon} />
      </Box>
    </>
  );

  const renderSuccessCard = () => (
    <>
      {isNew ? (
        <HStack
          pt='10px'
          color={`${COLORS.PINK}.650`}
        >
          <Icon as={BellIcon} />
          <Text>New</Text>
        </HStack>
      ) : null}
      <Box
        pos='absolute'
        bottom='84px'
        right='30px'
        data-testid='success'
      >
        <Icon as={SuccessIcon} />
      </Box>
    </>
  );

  const statusMapping = {
    initial: renderInitialCard,
    failed: renderFailedCard,
    success: renderSuccessCard
  };

  const renderFooterInitial = () => (
    <CardFooter
      px={convertPxToRem(30)}
      py={convertPxToRem(15)}
      data-testid='footer-initial'
    >
      <Box
        fontSize='12px'
        lineHeight='14px'
      >
        <Text>Created by</Text>
        <Text as='b'>{author}</Text>
        <Text>{dateTime}</Text>
      </Box>
    </CardFooter>
  );

  const renderFooterResult = () => (
    <CardFooter
      px={convertPxToRem(30)}
      py={convertPxToRem(15)}
      data-testid='footer-result'
    >
      <Box fontSize='12px'>
        <Text>Score:</Text>
        <HStack alignItems='baseline'>
          <Text
            fontSize='24px'
            fontWeight='700'
            lineHeight='29px'
          >
            {score}
          </Text>
          <Text
            fontSize='14px'
            lineHeight='17px'
          >
            /{maxScore}
          </Text>
        </HStack>
        <Text>{dateTime}</Text>
      </Box>
    </CardFooter>
  );

  const footerMapping = {
    initial: renderFooterInitial,
    failed: renderFooterResult,
    success: renderFooterResult
  };

  return (
    <Card
      colorScheme={color}
      h='350px'
      maxW={maxW}
      {...rest}
      pt={convertPxToRem(20)}
      pb={convertPxToRem(26)}
      data-testid='exam-card'
    >
      <CardHeader
        px={convertPxToRem(30)}
        py='0'
      >
        <HStack justify='space-between'>
          <Flex
            bg={COLORS.BLACK}
            color={`${color}.400`}
            w='43px'
            h='43px'
            borderRadius='full'
            align='center'
            justify='center'
            fontSize='25px'
          >
            {order}
          </Flex>
          <Flex
            bg={`${color}.500`}
            w='32px'
            h='32px'
            borderRadius='full'
            align='center'
            justify='center'
            fontSize='25px'
          >
            {icon}
          </Flex>
        </HStack>
      </CardHeader>
      <CardBody
        px={convertPxToRem(30)}
        pb={convertPxToRem(10)}
      >
        <Text
          fontSize='20px'
          lineHeight='25px'
          fontWeight='700'
        >
          {title}
        </Text>

        {statusMapping[status]()}
      </CardBody>
      {footerMapping[status]()}
    </Card>
  );
};

export default ExamCard;
