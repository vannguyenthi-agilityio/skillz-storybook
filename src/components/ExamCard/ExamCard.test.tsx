// Libraries
import { render, screen } from '@testing-library/react';

// Components
import ExamCard from '.';
import { SlackIcon } from '@components/Icons';

// Types
import { CARD_STATUS, COLORS } from '@enums';

const baseProps = {
  color: COLORS.CYAN,
  title: 'Lorem ipsum dolor sit amet, consectetu',
  status: CARD_STATUS.INITIAL,
  price: 2,
  author: 'Jame Rodriguez',
  dateTime: '8:00 AM - 21 Dec, 2022',
  enroller: 1024,
  score: 50,
  order: 1,
  icon: SlackIcon
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps
  };
  return render(<ExamCard {...props} />);
};

describe('Exam Card render', () => {
  test('should render correctly', () => {
    setup();
    const card = screen.getByTestId('exam-card');
    expect(card).toMatchSnapshot();
  });

  test('should render new success card', () => {
    setup({ isNew: true, status: CARD_STATUS.SUCCESS });
    const card = screen.getByText('New');
    expect(card).toBeInTheDocument();
  });

  test('should render success card', () => {
    setup({ status: CARD_STATUS.SUCCESS });
    const card = screen.getByTestId('success');
    expect(card).toBeTruthy();
  });

  test('should render new failed card', () => {
    setup({ isNew: true, status: CARD_STATUS.FAILED });
    const card = screen.getByText('New');
    expect(card).toBeInTheDocument();
  });

  test('should render failed card', () => {
    setup({ status: CARD_STATUS.FAILED });
    const card = screen.getByTestId('failed');
    expect(card).toBeTruthy();
  });

  test('should render initial footer', () => {
    setup({ status: CARD_STATUS.INITIAL });
    const card = screen.getByTestId('footer-initial');
    expect(card).toBeTruthy();
  });

  test('should render result footer', () => {
    setup({ status: CARD_STATUS.FAILED });
    const card = screen.getByTestId('footer-result');
    expect(card).toBeTruthy();
  });
});
