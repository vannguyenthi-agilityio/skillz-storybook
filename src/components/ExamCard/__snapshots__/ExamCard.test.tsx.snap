// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exam Card render should render correctly 1`] = `
<div
  class="chakra-card css-4sdy7u"
  data-testid="exam-card"
>
  <div
    class="chakra-card__header css-1835qdw"
  >
    <div
      class="chakra-stack css-1oeb4ru"
    >
      <div
        class="css-1shb4p1"
      >
        1
      </div>
      <div
        class="css-vvmdg6"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.9344 23.1378C13.6449 23.1335 12.6017 22.0873 12.6011 20.7978V18.4567H14.9344C16.2274 18.4567 17.2755 19.5048 17.2755 20.7978C17.2755 22.0907 16.2274 23.1389 14.9344 23.1389V23.1378ZM9.06555 23.1378C7.77606 23.1335 6.73282 22.0873 6.73222 20.7978V14.9378C6.73222 14.1028 7.17769 13.3312 7.90083 12.9137C8.62397 12.4962 9.51491 12.4962 10.238 12.9137C10.9612 13.3312 11.4067 14.1028 11.4067 14.9378V20.7978C11.406 22.0903 10.3581 23.1378 9.06555 23.1378ZM20.7878 17.2789H14.9344C13.6418 17.2789 12.5939 16.231 12.5939 14.9383C12.5939 13.6457 13.6418 12.5978 14.9344 12.5978H20.7878C22.0804 12.5978 23.1283 13.6457 23.1283 14.9383C23.1283 16.231 22.0804 17.2789 20.7878 17.2789ZM3.21333 17.2789C1.92067 17.2789 0.872772 16.231 0.872772 14.9383C0.872772 13.6457 1.92067 12.5978 3.21333 12.5978H5.54666V14.9378C5.54666 16.2299 4.49986 17.2777 3.20777 17.2789H3.21333ZM20.7878 11.4022H18.4544V9.06223C18.4581 7.77444 19.5044 6.73312 20.7922 6.73556C22.08 6.73802 23.1223 7.78332 23.1211 9.07111C23.1199 10.3589 22.0756 11.4022 20.7878 11.4022ZM14.9344 11.4022C13.6449 11.3979 12.6017 10.3517 12.6011 9.06223V3.20223C12.6011 2.36722 13.0466 1.59564 13.7697 1.17814C14.4929 0.760631 15.3838 0.760631 16.1069 1.17814C16.8301 1.59564 17.2755 2.36722 17.2755 3.20223V9.06223C17.2749 10.3548 16.227 11.4022 14.9344 11.4022ZM9.06555 11.4022H3.21333C1.92067 11.4022 0.872772 10.3543 0.872772 9.06168C0.872772 7.76902 1.92067 6.72112 3.21333 6.72112H9.06555C10.3582 6.72112 11.4061 7.76902 11.4061 9.06168C11.4061 10.3543 10.3582 11.4022 9.06555 11.4022ZM11.3989 5.54335H9.06555C8.11861 5.54604 7.26339 4.97775 6.89903 4.10371C6.53468 3.22966 6.73303 2.22218 7.4015 1.55148C8.06998 0.880771 9.07679 0.679071 9.95205 1.04051C10.8273 1.40196 11.3984 2.25528 11.3989 3.20223V5.54223V5.54335Z"
            fill="white"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="chakra-card__body css-1vys869"
  >
    <p
      class="chakra-text css-53hotv"
    >
      Lorem ipsum dolor sit amet, consectetu
    </p>
    <div
      class="chakra-stack css-gv9ozg"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 21.3184C17.1464 21.3184 21.3184 17.1464 21.3184 12C21.3184 6.85357 17.1464 2.68156 12 2.68156C6.85357 2.68156 2.68156 6.85357 2.68156 12C2.68156 17.1464 6.85357 21.3184 12 21.3184Z"
          fill="black"
        />
        <path
          clip-rule="evenodd"
          d="M24 12C24 18.6274 18.6274 24 12 24C5.37258 24 0 18.6274 0 12C0 5.37258 5.37258 0 12 0C18.6274 0 24 5.37258 24 12ZM22.6592 12C22.6592 17.8869 17.8869 22.6592 12 22.6592C6.11308 22.6592 1.34078 17.8869 1.34078 12C1.34078 6.11308 6.11308 1.34078 12 1.34078C17.8869 1.34078 22.6592 6.11308 22.6592 12Z"
          fill="black"
          fill-rule="evenodd"
        />
        <path
          d="M11.4846 6.80791C11.668 6.43628 12.1979 6.43628 12.3813 6.80791L13.6292 9.33644C13.7021 9.48402 13.8429 9.5863 14.0057 9.60997L16.7961 10.0154C17.2062 10.075 17.37 10.579 17.0732 10.8683L15.0541 12.8365C14.9362 12.9513 14.8824 13.1168 14.9103 13.279L15.3869 16.0582C15.457 16.4666 15.0283 16.7781 14.6614 16.5853L12.1656 15.2731C12.02 15.1966 11.846 15.1966 11.7003 15.2731L9.20448 16.5853C8.83766 16.7781 8.40895 16.4666 8.479 16.0582L8.95566 13.279C8.98348 13.1168 8.9297 12.9513 8.81186 12.8365L6.7927 10.8683C6.49595 10.579 6.6597 10.075 7.06981 10.0154L9.86022 9.60997C10.0231 9.5863 10.1639 9.48402 10.2367 9.33644L11.4846 6.80791Z"
          fill="#FFEBAB"
        />
      </svg>
      <p
        class="chakra-text css-1iqv7ej"
      >
        2
      </p>
    </div>
    <div
      class="chakra-stack css-84zodg"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 18 16"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.50002 0.5C4.19883 0.5 2.33335 2.36548 2.33335 4.66667C2.33335 6.96785 4.19883 8.83333 6.50002 8.83333C8.80121 8.83333 10.6667 6.96785 10.6667 4.66667C10.6667 2.36548 8.80121 0.5 6.50002 0.5ZM4.00002 4.66667C4.00002 3.28595 5.11931 2.16667 6.50002 2.16667C7.88073 2.16667 9.00002 3.28595 9.00002 4.66667C9.00002 6.04738 7.88073 7.16667 6.50002 7.16667C5.11931 7.16667 4.00002 6.04738 4.00002 4.66667Z"
          fill="#4D4D4D"
        />
        <path
          d="M13.0904 4.84857C12.8559 4.72903 12.5965 4.66672 12.3334 4.66672V3.00005C12.8597 3.00005 13.3785 3.12464 13.8474 3.36371C13.8991 3.39004 13.95 3.41768 14 3.44658C14.4045 3.68011 14.756 3.99641 15.0311 4.37546C15.3403 4.80143 15.5441 5.29456 15.6261 5.81447C15.708 6.33439 15.6656 6.86631 15.5025 7.36671C15.3393 7.86711 15.0599 8.32176 14.6873 8.69344C14.3146 9.06513 13.8593 9.34328 13.3584 9.50514C12.9128 9.64917 12.4424 9.69746 11.978 9.64767C11.9205 9.64151 11.8631 9.63384 11.8059 9.62467C11.2865 9.54143 10.7943 9.33648 10.3693 9.02658L10.3684 9.02594L11.3509 7.67964C11.5635 7.83476 11.8098 7.93731 12.0696 7.97895C12.3295 8.02059 12.5955 8.00011 12.8459 7.91919C13.0963 7.83826 13.324 7.69919 13.5103 7.51335C13.6966 7.32752 13.8363 7.1002 13.9179 6.85001C13.9994 6.59981 14.0206 6.33386 13.9797 6.07391C13.9387 5.81396 13.8368 5.56741 13.6822 5.35443C13.5276 5.14145 13.3248 4.9681 13.0904 4.84857Z"
          fill="#4D4D4D"
        />
        <path
          d="M15.6651 15.5C15.6651 15.0625 15.5789 14.6292 15.4115 14.225C15.2441 13.8208 14.9986 13.4535 14.6893 13.1441C14.3799 12.8347 14.0126 12.5893 13.6084 12.4219C13.2041 12.2544 12.7709 12.1682 12.3334 12.1682V10.5C12.9018 10.5 13.4653 10.5969 14 10.786C14.083 10.8153 14.1653 10.8468 14.2468 10.8806C14.8534 11.1319 15.4046 11.5002 15.8689 11.9645C16.3332 12.4288 16.7015 12.98 16.9528 13.5866C16.9865 13.6681 17.0181 13.7504 17.0474 13.8333C17.2364 14.368 17.3334 14.9316 17.3334 15.5H15.6651Z"
          fill="#4D4D4D"
        />
        <path
          d="M12.3334 15.5H10.6667C10.6667 13.1988 8.80121 11.3333 6.50002 11.3333C4.19883 11.3333 2.33335 13.1988 2.33335 15.5H0.666687C0.666687 12.2783 3.27836 9.66667 6.50002 9.66667C9.72168 9.66667 12.3334 12.2783 12.3334 15.5Z"
          fill="#4D4D4D"
        />
      </svg>
      <b
        class="chakra-text css-13pzcpe"
      >
        1024
      </b>
      <p
        class="chakra-text css-1e3rjkh"
      >
        already enrolled
      </p>
    </div>
  </div>
  <div
    class="chakra-card__footer css-1edy0g7"
    data-testid="footer-initial"
  >
    <div
      class="css-1puywlt"
    >
      <p
        class="chakra-text css-0"
      >
        Created by
      </p>
      <b
        class="chakra-text css-0"
      >
        Jame Rodriguez
      </b>
      <p
        class="chakra-text css-0"
      >
        8:00 AM - 21 Dec, 2022
      </p>
    </div>
  </div>
</div>
`;
